src/llmq/quorums_blockprocessor.cpp

```diff
@@ -271,6 +271,10 @@ void CQuorumBlockProcessor::UpgradeDB()
     if (chainActive.Height() >= Params().GetConsensus().DIP0003EnforcementHeight) {
         auto pindex = chainActive[Params().GetConsensus().DIP0003EnforcementHeight];
         while (pindex) {
+            if (fPruneMode && !(pindex->nStatus & BLOCK_HAVE_DATA)) {
```

Comment:
```suggestion
            if (fPruneMode && !(pindex->nStatus && BLOCK_HAVE_DATA)) {
```

Reply:
Nope, this is not a typo, it's an actual bitwise AND.

Reply:
I guess I don't understand the purpose of using a bitwise AND here

Reply:
It basically tests if a specific bit is set in an integer (which is used as a bitset). The result of the bitwise AND will only be non-zero when the bit specified by `BLOCK_HAVE_DATA` is also set in `nStatus`, and thanks to the strangeness of C/C++, this can be used like a boolean expression.